app/src/org/commcare/network/HttpRequestEndpointsMock.java

```diff
@@ -41,6 +41,11 @@ public HttpResponse makeCaseFetchRequest(String baseUri, boolean includeStateFla
     }
 
     @Override
+    public HttpResponse makeAsyncCaseFetchRequest() {
+        return HttpResponseMock.buildHttpResponseMockForAsyncRestore();
```

Comment:
Instead of this, I would check `LocalDataPullResponseFactory.INSTANCE.isAsyncRestore()` in `HttpRequestEndpointsMock.makeCaseFetchRequest`. 
It would be nice to avoid adding the `makeAsyncCaseFetchRequest` declaration to `HttpRequestEndpoints` just for testing purposes.